rust/src/datatypes.rs

```diff
@@ -280,6 +306,54 @@ impl Field {
         }
         None
     }
+
+    #[async_recursion]
+    async fn load_dictionary<'a>(&mut self, reader: &'a ObjectReader<'_>) -> Result<()> {
+        if let DataType::Dictionary(_, value_type) = self.data_type() {
+            assert!(self.dictionary.is_some());
+            if let Some(dict_info) = self.dictionary.as_mut() {
+                use DataType::*;
+                match value_type.as_ref() {
+                    Utf8 | Binary => {
+                        dict_info.values = Some(
+                            reader
+                                .read_binary_array(
+                                    value_type.as_ref(),
+                                    dict_info.offset,
+                                    dict_info.length,
+                                )
+                                .await?,
+                        );
+                    }
+                    Int8 | Int16 | Int32 | Int64 | UInt8 | UInt16 | UInt32 | UInt64 => {
+                        dict_info.values = Some(
+                            reader
+                                .read_primitive_array(
+                                    value_type.as_ref(),
+                                    dict_info.offset,
+                                    dict_info.length,
+                                )
+                                .await?,
+                        );
+                    }
+                    _ => {
+                        return Err(Error::Schema(format!(
+                            "Does not support {} as dictionary value type",
+                            value_type
+                        )));
+                    }
+                }
+            } else {
+                panic!("Should not reach here: dictionary field does not load dictionary info")
```

Comment:
same here. Err or panic! ? when to use what?

Reply:
feel it is a bug? much like "assert" in C++?